platform/ios/src/MGLMapView.mm

```diff
@@ -804,222 +817,137 @@ + (BOOL)requiresConstraintBasedLayout
+    _logoViewPosition = logoViewPosition;
+    [self installLogoViewConstraints];
+}
+
+- (void)setLogoViewOffset:(CGPoint)logoViewOffset {
+    NSAssert(CGRectContainsPoint(self.bounds, logoViewOffset), @"The position offset of the logo should within the mapview boundaries.");
+    MGLLogDebug(@"Setting logoViewOffset: (x:%f, y:%f)", logoViewOffset.x, logoViewOffset.y);
+    _logoViewOffset = logoViewOffset;
+    [self installLogoViewConstraints];
+}
+
+- (void)setAttributionButtonPosition:(MGLOrnamentPosition)attributionButtonPosition {
+    MGLLogDebug(@"Setting attributionButtonPosition: %lu", attributionButtonPosition);
+    _attributionButtonPosition = attributionButtonPosition;
+    [self installAttributionButtonConstraints];
+}
+
+- (void)setAttributionButtonOffset:(CGPoint)attributionButtonOffset {
+    NSAssert(CGRectContainsPoint(self.bounds, attributionButtonOffset), @"The position offset of the attribution should within the mapview boundaries.");
+    MGLLogDebug(@"Setting attributionButtonOffset: (x:%f, y:%f)", attributionButtonOffset.x, attributionButtonOffset.y);
+    _attributionButtonOffset = attributionButtonOffset;
+    [self installAttributionButtonConstraints];
+}
+
+- (void)updateConstraintsForOrnament:(UIView *)view
+                       constraints:(NSMutableArray *)constraints
+                          position:(MGLOrnamentPosition)position
+                              size:(CGSize)size
+                            offset:(CGPoint)offset {
+    NSMutableArray *updatedConstraints = [NSMutableArray array];
     
-    [super updateConstraints];
+    switch (position) {
+        case MGLOrnamentPositionTopLeft:
+            [updatedConstraints addObject:[view.topAnchor constraintEqualToAnchor:self.mgl_safeTopAnchor constant:offset.y]];
+            [updatedConstraints addObject:[view.leadingAnchor constraintEqualToAnchor:self.mgl_safeLeadingAnchor constant:offset.x]];
+            break;
+        case MGLOrnamentPositionTopRight:
+            [updatedConstraints addObject:[view.topAnchor constraintEqualToAnchor:self.mgl_safeTopAnchor constant:offset.y]];
+            [updatedConstraints addObject:[self.mgl_safeTrailingAnchor constraintEqualToAnchor:view.trailingAnchor constant:offset.x]];
+            break;
+        case MGLOrnamentPositionBottomLeft:
+            [updatedConstraints addObject:[self.mgl_safeBottomAnchor constraintEqualToAnchor:view.bottomAnchor constant:offset.y]];
+            [updatedConstraints addObject:[view.leadingAnchor constraintEqualToAnchor:self.mgl_safeLeadingAnchor constant:offset.x]];
+            break;
+        case MGLOrnamentPositionBottomRight:
+            [updatedConstraints addObject:[self.mgl_safeBottomAnchor constraintEqualToAnchor:view.bottomAnchor constant:offset.y]];
+            [updatedConstraints addObject: [self.mgl_safeTrailingAnchor constraintEqualToAnchor:view.trailingAnchor constant:offset.x]];
+            break;
+        default:
```

Comment:
Since we control the possible cases `MGLOrnamentPosition`, don’t handle `default` here. If someone adds a new case and leaves it unhandled, Xcode will warn that developer that this switch is not exhaustive (and hopefully they’ll fix it).